### PR TITLE
Upgrading to SonarQube API 6.7

### PR DESCRIPTION
**Pull request by redLeR1st**
Updated:  

- All pom.xml files

Replaced: 
- 'Settings' by 'Configuration' because 'Settings' were deprecated in 6.5 API
- 'ProfileDefinition' by 'BuiltInQualityProfilesDefinition' because
	          'BuiltInQualityProfilesDefinition' were deprecated in 6.6 API

Removed: 
- 'Initializer' because this extension point is deprecated

### CODE REVIEW

File: src/sonarqube-analyzers/sourcemeter-analyzer-base/src/main/java/com/sourcemeter/analyzer/base/core/VersionChecker.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ public void onServerStart(Server server) {
         int majorVersion = Integer.parseInt(version[0]);
         int minorVersion = Integer.parseInt(version[1]);
 
-        if (!(majorVersion == 6 && minorVersion >= 3)) {
+        if (!(majorVersion  >= 6 && minorVersion >= 0)) {
```
Comment (LaszloLango) [inline src/sonarqube-analyzers/sourcemeter-analyzer-base/src/main/java/com/sourcemeter/analyzer/base/core/VersionChecker.java]: This will accept all of the 6.x versions, but we only supports 6.7+

Comment (redLeR1st) [reply src/sonarqube-analyzers/sourcemeter-analyzer-base/src/main/java/com/sourcemeter/analyzer/base/core/VersionChecker.java]: Resolved

File: src/sonarqube-analyzers/sourcemeter-analyzer-csharp/src/main/java/com/sourcemeter/analyzer/csharp/profile/SourceMeterCSharpProfile.java [hunk 2]
```diff
@@ -30,33 +30,30 @@
...
+
+        super.parseXml(qualityProfile, qualityProfileXml);
+
         try {
-            input = getClass().getResourceAsStream(
-                    "/SourceMeter_way_default_profile.xml");
-            return super.createProfile(input, ValidationMessages.create());
-        } finally {
-            IOUtils.closeQuietly(input);
+            qualityProfileXml.close();
```
Comment (pille5) [inline src/sonarqube-analyzers/sourcemeter-analyzer-csharp/src/main/java/com/sourcemeter/analyzer/csharp/profile/SourceMeterCSharpProfile.java]: Not just the `close` can throw `IOException`. Why did you remove the `finally` block?

Comment (redLeR1st) [reply src/sonarqube-analyzers/sourcemeter-analyzer-csharp/src/main/java/com/sourcemeter/analyzer/csharp/profile/SourceMeterCSharpProfile.java]: Resolved for all five analyzers

**Comment by pille5** [comment]
@redLeR1st please rebase to the master.
